io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriter.java

```diff
@@ -1530,6 +1562,14 @@ private void writeElementValue(Object value, PropertyDefinition propDef)
 							propType.getConstraint(ElementType.class).getDefinition()));
 				}
 			}
+			else if (value instanceof Double || value instanceof Float
+					|| value instanceof BigDecimal) {
+				// Apply formatting only to decimal values, not integers
+				String representation = DecimalFormatUtil.applyFormatter((Number) value,
```

Comment:
I think it would be better to continue using the XML specific conversion if no format is configured (instead of just using `String.valueOf(...)`).